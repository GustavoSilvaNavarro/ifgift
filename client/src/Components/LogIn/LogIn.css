.login-container {
  background-image: url('../../../public/loginbg.jpg');
  background-position: center;
  background-size: cover;
  display: flex;
  height: 100vh;
  width: 100vw;
  justify-content: center;
  align-items: center;
  z-index: 201;
}

.login-box {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: white;
  border: 5px solid #faedcb;
  width: 80%;
  max-width: 600px;
  height: 50%;
  border-radius: 6px;
  box-shadow: 0 10px 25px -5px rgba(0,0,0,0.2);
}


.to-box {
  display: flex;
  width: 80%;
  justify-content: flex-start;
  align-items: center;
  margin-left: 4rem;
}

.from-box {
  display: flex;
  width: 75%;
  justify-content: flex-start;
  align-items: center;
}

.logo {
  height: 3rem;
  margin-left: -5%;
}

.you {
  display: flex;
  font-family:'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  margin-left: 2rem;
}

.question {
  margin-top: 1rem;
}

.function-list {
  display: flex;
  flex-direction: column;
  /* justify-content: center;
  align-items: center; */
  width: 50%;
  margin-top: 0.5rem;
  margin-left: 1.5rem;
}

.login-btn {
  background-color: #dbcdf0;
  display: flex;
  padding: 0.7rem;
  border-radius: 6px;
  border: none;
  margin-top: 5%;
  box-shadow: 0 10px 25px -5px rgba(0,0,0,0.4);
}

.ifgift-copy {
  display: flex;
  margin-top: 1.5rem;
}