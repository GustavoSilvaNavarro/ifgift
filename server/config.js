module.exports = {
  dbName: '',
  PORT: 3333
};