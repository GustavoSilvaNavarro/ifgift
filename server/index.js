const Koa = require('koa');
const app = new Koa();
const bodyParser = require('koa-bodyparser');
const cors = require('@koa/cors');

const config = require('./config');
const router = require('./router');

app.use(bodyParser());
app.use(cors());
app.use(router.routes());

app.listen(config.PORT, () => {
  console.log(`Server kickin it at http://localhost:${config.PORT}`);
})