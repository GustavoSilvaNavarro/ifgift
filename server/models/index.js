const mongoose = require('mongoose');
const config = require('../config');

mongoose.connect(`mongodb://localhost:27017/${config.dbName}`).then(
  () => { console.log('Database is connected.'); },
  err => { console.error(err); }
);

module.exports = mongoose;