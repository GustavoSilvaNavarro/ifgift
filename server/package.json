{
  "name": "if-gift-server",
  "version": "1.0.0",
  "description": "Legacy Project Server",
  "main": "dist/app.js",
  "scripts": {
    "start": "node .",
    "dev:server": "nodemon src/app.ts --exec ts-node",
    "lint:server": "eslint \"src/**/*.{js,jsx,ts,tsx}\" --quiet",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "build": "tsc",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "test:watch": "jest --watch"
  },
  "keywords": [],
  "author": "Gustavo Silva | James Warnsby",
  "license": "ISC",
  "dependencies": {
    "@koa/cors": "^4.0.0",
    "@koa/router": "^12.0.0",
    "jsonwebtoken": "^8.5.1",
    "koa": "^2.13.4",
    "koa-body": "^6.0.1",
    "koa-router": "^12.0.0",
    "mongoose": "^6.7.2",
    "nodemon": "^2.0.20"
  },
  "devDependencies": {
    "@faker-js/faker": "^7.6.0",
    "@types/jest": "^29.2.3",
    "@types/jsonwebtoken": "^8.5.9",
    "@types/koa": "^2.13.5",
    "@types/koa__cors": "^3.3.0",
    "@types/koa__router": "^12.0.0",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.43.0",
    "@typescript-eslint/parser": "^5.43.0",
    "dotenv": "^16.0.3",
    "eslint": "^8.27.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-standard-with-typescript": "^23.0.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "^27.1.5",
    "eslint-plugin-n": "^15.5.1",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-promise": "^6.1.1",
    "jest": "^29.3.1",
    "prettier": "^2.7.1",
    "supertest": "^6.3.1",
    "ts-jest": "^29.0.3",
    "ts-node": "^10.9.1",
    "typescript": "^4.9.3"
  }
}
