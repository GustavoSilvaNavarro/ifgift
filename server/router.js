const Router = require('koa-router');
const router = new Router();
const profile = require('./controllers/profile');
const lists = require('./controllers/lists')

router.get('/profile', profile.getProfileInfo);
router.post('/profile', profile.insertProfile);

router.get('/mylists', lists.fetchMyLists);
router.post('/mylists', lists.insertList);
router.delete('/mylists');

module.exports = router;
