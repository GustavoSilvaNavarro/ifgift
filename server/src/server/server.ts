import Koa from 'koa';
import bodyParser from 'koa-bodyparser';
import cors from '@koa/cors';

import userRoutes from '../routes/user-routes/user-routes';

const app: Koa = new Koa();
const PORT = process.env.PORT || 8080;

app.use(bodyParser());
app.use(cors());
app.use(userRoutes.routes());
app.use(userRoutes.allowedMethods());

export default { app, PORT };
