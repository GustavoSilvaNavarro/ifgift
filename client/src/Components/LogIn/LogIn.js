import React, { useEffect } from "react";
import { useAuth0 } from "@auth0/auth0-react";

import './LogIn.css';

import { useNavigate } from "react-router";

const LogIn = () => {
  const navigate = useNavigate();
  const { loginWithPopup, isAuthenticated } = useAuth0();

  useEffect(() => {
    if (isAuthenticated) {
      navigate('/home');
    }
  }, [navigate, isAuthenticated]);

  return (
    <div className="login-container">
      <div className="login-box">
        <h2>From:</h2>
        <h2>To: You!</h2>
        <p>Don't know what to get that special someone?</p>
        <ul>
          <li>Check their ifGift want list!</li>
          <li>Or keep your own lists of gifting ideas for all the special folks in your life.</li>
        </ul>
        <p>Dive in to a better way of giving with ifGift.</p>
        <button
          onClick={loginWithPopup}
          type="button"
          className="login-btn"
        >
          Sign up
        </button>
      </div>
    </div>
  )
};

export default LogIn;