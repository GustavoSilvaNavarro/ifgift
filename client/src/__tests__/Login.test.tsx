import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom/server';
import { useAuth0, User } from '@auth0/auth0-react';

import { LogIn } from '../Components/LogIn/LogIn';

jest.mock('@auth0/auth0-react');

interface MockedUseAuth0 extends User {
  mockReturnValue: jest.Mock;
}

describe('Login Process using Auth0', () => {
  test('Login Process using Auth0', () => {
    const loginWithPopup = jest.fn();
    (useAuth0 as unknown as MockedUseAuth0).mockReturnValue({
      loginWithPopup,
    });

    render(
      <StaticRouter location={'/'}>
        <LogIn />
      </StaticRouter>
    );

    const img = screen.getByAltText('ifGift logo');
    const loginBtn = screen.getByRole('button', { name: 'Sign up' });
    userEvent.click(loginBtn);

    expect(img).toBeInTheDocument();
    expect(useAuth0).toHaveBeenCalled();
    expect(loginWithPopup).toHaveBeenCalled();
  });
});

describe('Login Component tests', () => {
  test('Render Correctly Login Component', () => {
    const loginWithPopup = jest.fn();
    (useAuth0 as unknown as MockedUseAuth0).mockReturnValue({
      loginWithPopup,
    });

    const tree = create(
      <StaticRouter location={'/'}>
        <LogIn />
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Login Button displays text', () => {
    const loginWithPopup = jest.fn();
    (useAuth0 as unknown as MockedUseAuth0).mockReturnValue({
      loginWithPopup,
    });

    render(
      <StaticRouter location={'/'}>
        <LogIn />
      </StaticRouter>
    );

    const loginBtn = screen.getByRole('button');
    expect(loginBtn).toHaveTextContent(/Sign up/);
  });
});
