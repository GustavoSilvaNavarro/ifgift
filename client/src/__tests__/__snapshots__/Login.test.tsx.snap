// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component tests Render Correctly Login Component 1`] = `
<div
  className="login-container"
>
  <div
    className="login-box"
  >
    <div
      className="from-box"
    >
      <h2>
        From:
      </h2>
      <img
        alt="ifGift logo"
        className="logo"
        src="https://static.wixstatic.com/media/3dbed1_4a14f7c810ff4d40ae1be96c5c171caa~mv2.png/v1/fill/w_541,h_168,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/ifGift.png"
      />
    </div>
    <div
      className="to-box"
    >
      <h2>
        To:
      </h2>
      <h2
        className="you"
      >
        You!
      </h2>
    </div>
    <p
      className="question"
    >
      Don't know what to get that special someone?
    </p>
    <ul
      className="function-list"
    >
      <li>
        Check their ifGift want list!
      </li>
      <li>
        Keep your own lists of gifting ideas for all the special folks in your life.
      </li>
    </ul>
    <p
      className="ifgift-copy"
    >
      Unwrap the better way to give.
    </p>
    <button
      className="login-btn"
      onClick={[Function]}
      type="button"
    >
      Sign up
    </button>
  </div>
</div>
`;
